contrib/transaction_profiling_analyzer/transaction_profiling_analyzer.py

```diff
@@ -514,76 +516,39 @@ def __init__(self, k):
         self.k = k
         from sortedcontainers import SortedDict
         self.ranges = SortedDict()
+        self.ranges[b''] = 0
 
     def process(self, transaction_info):
         for get_range in transaction_info.get_ranges:
             self._insert_range(get_range.key_range.start_key, get_range.key_range.end_key)
 
     def _insert_range(self, start_key, end_key):
         keys = self.ranges.keys()
-        if len(keys) == 0:
-            self.ranges[start_key] = end_key, 1
-            return
+        start_pos = bisect_right(keys, start_key)-1
```

Comment:
We might want to do a little more testing of this part. The work done in this block is what causes the run time to be long as we expand the time window (the cost goes up with each range mutation inserted), and I've seen performance be better and worse depending on where and when I ran it.

If we could change the performance of this algorithm substantially, it would make a big difference in the how quickly this tool can run and how wide of a window we can measure.

Reply:
With my second commit, I think the new version is strictly faster.